rust/ql/src/queries/unusedentities/UnreachableCode.ql

```diff
@@ -0,0 +1,59 @@
+/**
+ * @name Unreachable code
+ * @description Code that cannot be reached should be deleted.
+ * @kind problem
+ * @problem.severity recommendation
+ * @precision medium
+ * @id rust/dead-code
+ * @tags maintainability
+ */
+
+import rust
+import codeql.rust.controlflow.ControlFlowGraph
+import codeql.rust.controlflow.internal.ControlFlowGraphImpl as ControlFlowGraphImpl
```

Comment:
I think queries should avoid using `internal` implementation classes. Is the public API insufficient somehow?

Reply:
As I understand it the public CFG API only exposes reachable nodes, which makes it difficult to reason about unreachable nodes without looking at the construction.  I agree its not ideal and I'm open to suggestions.

Reply:
Ah, yes of course.